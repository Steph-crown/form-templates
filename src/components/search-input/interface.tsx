export interface ISearchInput {
    placeholder: string;
    value: string;
    name: string;
    width?: string;

    // Accepts a react change event and returns nothing
    handleChange?: (e: React.ChangeEvent<any>) => void;

    // Clears the content of the input
    handleClear: (e: React.ChangeEvent<any>) => void;
}
