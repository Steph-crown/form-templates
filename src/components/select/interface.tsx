export interface ISelectProps {
    options: ISelectOption[];
    label: string;
    placeholder: string;
    disabled?: boolean;
    width?: string;
    handleChange: any;
    selectedItem: { label: string; value: string | number } | undefined;
    className?: string;
}

export interface ISelectOption {
    label: string;
    value: string;
}
