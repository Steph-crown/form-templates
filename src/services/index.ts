import { ITemplateData } from "./services.interface";
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { API_URL } from "../consts/api-url";

/**
 * API slice
 */
export const api = createApi({
    reducerPath: "api",

    baseQuery: fetchBaseQuery({
        baseUrl: API_URL,
    }),

    endpoints: (bulder) => ({
        // Get all templates
        getFormTemplates: bulder.query<ITemplateData[], void>({
            query: () => ({ url: "", method: "GET" }),
        }),
    }),
});

export const { useGetFormTemplatesQuery } = api;
