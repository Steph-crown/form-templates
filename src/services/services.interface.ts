export interface ITemplateData {
    name: string;
    created: string;
    category: string[];
    description: string;
    link: string;
}

// [
//       Template {
//                  name: string <Template name>,
//                  created: string <Template creation date in iso format>,
//                  category: array[string] <Categories assigned to the template>
//                  description: string <Template description>
//                  link: string <Template link>
//               }
//  ]
