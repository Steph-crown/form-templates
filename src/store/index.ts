import { api } from "./../services/index";
import { configureStore } from "@reduxjs/toolkit";

import formTemplatesReducer from "./../slices/form-templates.slice";

export const store = configureStore({
    reducer: {
        [api.reducerPath]: api.reducer,

        // Reducer for the form templates
        formTemplates: formTemplatesReducer,
    },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;

// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
