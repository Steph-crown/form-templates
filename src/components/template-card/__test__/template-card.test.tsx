import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { TemplateCard } from "..";
import { store } from "../../../store";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ITemplateData } from "../../../services/services.interface";

afterEach(cleanup);
const testData: ITemplateData = {
    name: "A template",
    created: "2022-02-14T17:42:11.521444",
    category: ["Education"],
    description: "A template for education",
    link: "https://www.google.com",
};

// Tests that the component renders without crashing
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <Provider store={store}>
            <TemplateCard {...testData} />
        </Provider>,
        div
    );
});

// Checks if the name rendered is correct
it("renders template name correctly", () => {
    const { getByTestId } = render(
        <Provider store={store}>
            <TemplateCard {...testData} />
        </Provider>
    );
    expect(getByTestId("template-name")).toHaveTextContent("A template");
});

// Checks snapshot of the component.
it("matches snapshot", () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <TemplateCard {...testData} />{" "}
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
