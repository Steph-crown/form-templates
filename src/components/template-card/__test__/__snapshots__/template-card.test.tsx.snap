// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <li
    className="sc-bdvvtL cWuhCu"
  >
    <div>
      <h2
        data-testid="template-name"
      >
        A template
      </h2>
      <p>
        A template for education
      </p>
    </div>
    <button>
      Use Template
    </button>
  </li>,
  " ",
]
`;
